edgePy/DGEList.py

```diff
@@ -98,16 +109,31 @@ def counts(self, counts: np.ndarray) -> np.matrix:
             counts: Columns correspond to samples and row to genes.
 
         """
+        if counts is None:
+            self._counts = None
+            return
+
+        if hasattr(self, '_counts'):
+            # do checks for things here.  You shouldn't modify counts
+            # if it has already been set.  Create a new obj.
+            if hasattr(self, '_samples') and self._samples is not None:
+                sample_count, gene_count = counts.shape
+                if sample_count != self._samples.shape[0] or gene_count != self._genes.shape[0]:
```

Comment:
This validation should be performed after we assert the type of `counts`,not before. Move code block to line ~135.

You can simplify to:

```python
if self.counts is not None and self.counts.shape != counts.shape:
    raise ValueError('..')
```